### DISCUSSION TITLE
Validate Dates

### DISCUSSION THREAD

**Discussion by Julian1993ARG** [serenity-is/Serenity#6987] [Help] [answered]
Hi, I need to validate dates in a dialog. Currently, I am using addValidationRule, but when the user changes the input, it gives me the last value.

Here is my code, which is in the constructor:

``` C#
this.form.FromDate.addValidationRule(this.uniqueName, e => {
            const date = this.form.FromDate.valueAsDate;
            if (!date) return "Invalid Date";
            if (date <= new Date()) return "The date should be after today"
        });
```

If the user selects a date before today, the validation doesn't work. However, if the user then selects another date, the validation works with the last value.

**Comment by luongnv6** [comment, marked as answer]
try set minDate and maxDate:
```ts
//set minDate
this.form.FromDate.element.datepicker("option", "minDate", new Date());
this.form.FromDate.set_minDate(new Date());

//or set maxDate
this.form.FromDate.element.datepicker("option", "maxDate", new Date());
this.form.FromDate.set_maxDate(new Date());
```

**Reply by Julian1993ARG** [reply]
Thanks, that worked for me!!
